subsys/bluetooth/services/cts_c.c

```diff
@@ -0,0 +1,251 @@
+	const struct bt_gatt_dm_attr *gatt_chrc;
+	const struct bt_gatt_dm_attr *gatt_desc;
+
+	if (bt_uuid_cmp(gatt_service->uuid, BT_UUID_CTS)) {
+		return -ENOTSUP;
+	}
+	LOG_DBG("CTS found");
+
+	gatt_chrc = bt_gatt_dm_char_by_uuid(dm, BT_UUID_CTS_CURRENT_TIME);
+	if (gatt_chrc == NULL) {
+		return -EINVAL;
+	}
+	gatt_desc = bt_gatt_dm_desc_by_uuid(dm, gatt_chrc,
+					    BT_UUID_CTS_CURRENT_TIME);
+	if (gatt_desc == NULL) {
+		return -EINVAL;
+	}
+	cts_c->handle_ct = gatt_desc->handle;
+	gatt_desc = bt_gatt_dm_desc_by_uuid(dm, gatt_chrc, BT_UUID_GATT_CCC);
+	if (gatt_desc == NULL) {
+		return -EINVAL;
+	}
+	cts_c->handle_ct_ccc = gatt_desc->handle;
+	LOG_DBG("Current Time characteristic found.");
+
+	/* Finally - save connection object */
+	cts_c->conn = bt_gatt_dm_conn_get(dm);
+
+	return 0;
+}
+
+/**@brief Function for decoding a read from the Current Time characteristic.
+ *
+ * @param[in] p_time  Current Time structure.
+ * @param[in] p_data  Pointer to the buffer containing the Current Time.
+ * @param[in] length  Length of the buffer containing the Current Time.
+ *
+ * @retval 0        If the time struct is valid.
+ * @retval -EINVAL 	If the length does not match the expected size of the data.
+ */
+static int current_time_decode(current_time_char_t *p_time,
+			       uint8_t const *p_data, uint32_t const length)
+{
+	if (length != CTS_C_CURRENT_TIME_EXPECTED_LENGTH) {
+		/* Return to prevent accessing the out-of-bounds data. */
+		return -EINVAL;
+	}
+
+	uint32_t index = 0;
+
+	/* Date. */
+	p_time->exact_time_256.year = sys_get_le16(p_data + index);
+	index += sizeof(uint16_t);
+	p_time->exact_time_256.month = p_data[index++];
+	p_time->exact_time_256.day = p_data[index++];
+	p_time->exact_time_256.hours = p_data[index++];
+	p_time->exact_time_256.minutes = p_data[index++];
+	p_time->exact_time_256.seconds = p_data[index++];
+
+	/* Day of week. */
+	p_time->exact_time_256.day_of_week = p_data[index++];
+
+	/* Fractions of a second. */
+	p_time->exact_time_256.fractions256 = p_data[index++];
+
+	/* Reason for updating the time. */
+	p_time->adjust_reason.manual_time_update = (p_data[index] >> 0) & 0x01;
+	p_time->adjust_reason.external_reference_time_update =
+		(p_data[index] >> 1) & 0x01;
+	p_time->adjust_reason.change_of_time_zone = (p_data[index] >> 2) & 0x01;
+	p_time->adjust_reason.change_of_daylight_savings_time =
+		(p_data[index] >> 3) & 0x01;
+
+	return 0;
+}
+
+/**@brief Function for sanity check of a Current Time structure.
+ *
+ * @param[in] p_time  Current Time structure.
+ *
+ * @retval 0        If the time struct is valid.
+ * @retval -EINVAL  If the time is out of bounds.
+ */
+static int current_time_validate(current_time_char_t *p_time)
+{
+	/* Year. */
+	if ((p_time->exact_time_256.year > CTS_YEAR_MAX) ||
+	    ((p_time->exact_time_256.year < CTS_YEAR_MIN) &&
+	     (p_time->exact_time_256.year != 0))) {
+		return -EINVAL;
+	}
+
+	/* Month. */
+	if (p_time->exact_time_256.month > 12) {
```

Comment:
Please define this magic number for this function. Since all this checks return the same error, I think it can be grouped in one if statement

Reply:
Done. Please note the magic number is defined in the start of file.